### ISSUE TITLE
[CPP]: Add query for CWE-200 Exposure of Sensitive Information to an Unauthorized Actor 

### ISSUE THREAD

**Issue by ihsinme** [github/securitylab#549]
### Query PR

https://github.com/github/codeql/pull/6950

### Language

C/C++

### CVE(s) ID list

- [CVE-2013-7458](https://nvd.nist.gov/vuln/detail/CVE-2013-7458)
- [CVE-2018-19044](https://nvd.nist.gov/vuln/detail/CVE-2018-19044)
- [CVE-2013-1813](https://nvd.nist.gov/vuln/detail/CVE-2013-1813)
- [CVE-2016-6494](https://nvd.nist.gov/vuln/detail/CVE-2016-6494)

### CWE

		 CWE-200: Exposure of Sensitive Information to an Unauthorized Actor

### Report

1. Since the rights are not set, it is possible to access files created or used by the program for a wide range of users.
2. The user who has access to the file can either read the sensitive information or change the input data to the program.
3. search for work with a file without setting access rights.
4. I completely exclude the presence of functions for setting rights anywhere in the program.

### Are you planning to discuss this vulnerability submission publicly? (Blog Post, social networks, etc).

- [X] Yes
- [ ] No

### Blog post link

https://twitter.com/ihsinme/status/1501064729623728131

**Comment by m-y-mo** [comment]
@ihsinme 
Thanks for the submission. After some discussions with my colleagues, we arrived at the following conclusion. This query tries to catch the use of `fopen` without setting file permission afterwards. This is a very common usage of `fopen` and in most cases it is not a mistake. There are different problems with this query, but one of the biggest problem is that, in most uses of `fopen`, it either read or write to an existing file rather than creating a new one, and there is no way to tell from the code whether this is the case or not. When `fopen` does not create a new file, then the query is flagging false positives, to which there are a lot of them. Even in the case where a file is created, there is the assumption that the file is sensitive. This again, are almost impossible to catch from the query. Another issue with using `chmod` after the creation of a file is that the code will still be subject to race condition and TOCTOU attacks, so we believe that while the issue may lead to security problem, it would be better addressed via proper configurations in the platform rather than CodeQL queries. As a result, I'm afraid that while we acknowledged your contribution to the experimental query suite via pull request, we are unable to accept the query for bounty. Thank you very much for your submission and contribution via pull request.
